tests/unit/query_v2_expression_evaluator.cpp

```diff
@@ -0,0 +1,1786 @@
+  ASSERT_EQ(EvaluateFunction("TOINTEGER", false).ValueInt(), 0);
+  ASSERT_EQ(EvaluateFunction("TOINTEGER", true).ValueInt(), 1);
+  ASSERT_EQ(EvaluateFunction("TOINTEGER", "\n\t3").ValueInt(), 3);
+  ASSERT_EQ(EvaluateFunction("TOINTEGER", " -3.5 \n\t").ValueInt(), -3);
+  ASSERT_TRUE(EvaluateFunction("TOINTEGER", "\n\t3X ").IsNull());
+  ASSERT_EQ(EvaluateFunction("TOINTEGER", -3.5).ValueInt(), -3);
+  ASSERT_EQ(EvaluateFunction("TOINTEGER", 3.5).ValueInt(), 3);
+}
+
+TEST_F(FunctionTest, Type) {
+  ASSERT_THROW(EvaluateFunction("TYPE"), FunctionRuntimeException);
+  ASSERT_TRUE(EvaluateFunction("TYPE", TypedValue()).IsNull());
+  auto e = CreateEdge({}, shard_manager.get());
+  ASSERT_EQ(EvaluateFunction("TYPE", e).ValueString(), "edge_type");
+  ASSERT_THROW(EvaluateFunction("TYPE", 2), FunctionRuntimeException);
+}
+
+TEST_F(FunctionTest, ValueType) {
+  ASSERT_THROW(EvaluateFunction("VALUETYPE"), FunctionRuntimeException);
+  ASSERT_THROW(EvaluateFunction("VALUETYPE", TypedValue(), TypedValue()), FunctionRuntimeException);
+  ASSERT_EQ(EvaluateFunction("VALUETYPE", TypedValue()).ValueString(), "NULL");
+  ASSERT_EQ(EvaluateFunction("VALUETYPE", TypedValue(true)).ValueString(), "BOOLEAN");
+  ASSERT_EQ(EvaluateFunction("VALUETYPE", TypedValue(1)).ValueString(), "INTEGER");
+  ASSERT_EQ(EvaluateFunction("VALUETYPE", TypedValue(1.1)).ValueString(), "FLOAT");
+  ASSERT_EQ(EvaluateFunction("VALUETYPE", TypedValue("test")).ValueString(), "STRING");
+  ASSERT_EQ(
+      EvaluateFunction("VALUETYPE", TypedValue(std::vector<TypedValue>{TypedValue(1), TypedValue(2)})).ValueString(),
+      "LIST");
+  ASSERT_EQ(EvaluateFunction("VALUETYPE", TypedValue(std::map<std::string, TypedValue>{{"test", TypedValue(1)}}))
+                .ValueString(),
+            "MAP");
+  auto v1 = CreateVertex({}, shard_manager.get());
+  ASSERT_EQ(EvaluateFunction("VALUETYPE", v1).ValueString(), "NODE");
+  auto e = CreateEdge({}, shard_manager.get());
+  ASSERT_EQ(EvaluateFunction("VALUETYPE", e).ValueString(), "RELATIONSHIP");
+  // TODO(kostasrim) Fix this when we add Path to accessors
+  //  Path p(v1, *e, v2);
+  //  ASSERT_EQ(EvaluateFunction("VALUETYPE", p).ValueString(), "PATH");
+}
+
+TEST_F(FunctionTest, Labels) {
+  ASSERT_THROW(EvaluateFunction("LABELS"), FunctionRuntimeException);
+  ASSERT_TRUE(EvaluateFunction("LABELS", TypedValue()).IsNull());
+  Label label{shard_manager->NameToLabel("label1")};
+  auto v = CreateVertex({}, shard_manager.get(), std::move(label));
+  std::vector<std::string> labels;
+  auto _labels = EvaluateFunction("LABELS", v).ValueList();
+  labels.reserve(_labels.size());
+  for (auto label : _labels) {
+    labels.emplace_back(label.ValueString());
+  }
+  ASSERT_THAT(labels, UnorderedElementsAre("label1"));
+  ASSERT_THROW(EvaluateFunction("LABELS", 2), FunctionRuntimeException);
+}
+
+// TODO(kostasrim) Enable this once we fix accessors Path
+// TEST_F(FunctionTest, NodesRelationships) {}
```

Comment:
Please remove that commented out function body. Also, if we cannot test a function, isn't it better to remove the whole function?